binary/script/embed.ipxe

```diff
@@ -6,4 +6,23 @@ echo Welcome to Neverland!
 # Allow the operator to drop to a shell
 prompt --key 0x02 --timeout 2000 Press Ctrl-B for the iPXE command line... && shell ||
 
+# This is possible because the DHCP options from the original vendor PXE DHCP request
+# are available to chainloaded iPXE binaries. See https://github.com/ipxe/ipxe/issues/188
+set vlan-id ${43.116:string}
+isset ${vlan-id} && goto boot-with-vlan ||
+
+:autoboot
 autoboot
+
+:boot-with-vlan
+set idx:int32 0
+# Find the first interface that is configured
+:loop isset ${net${idx}/mac} && goto loop_done ||
```

Comment:
Did you by any chance test this on a machine with multiple NICs and verify that it behaves correctly booting from all of them?

To my eye I'd expect this to find the first NIC with a valid MAC address.

I wonder if there are any other hints from the prior DHCP request that could be helpful here?

I'm mostly flagging this as a place to look for bugs later. I think we should land this so that further testing can proceed. It's already a big step up.

Reply:
another great point! yes, i did test with multiple network interfaces. I found that when the PXE firmware would request PXE it would show trying against, for example, `net0`, `net1`, etc but once our iPXE binary was loaded, i found that only `net0` was available. That was my very specific testing.